src/backend/distributed/operations/repair_shards.c

```diff
@@ -287,7 +429,120 @@ citus_move_shard_placement(PG_FUNCTION_ARGS)
 
 
 /*
- * master_move_shard_placement is a wrapper function for old UDF name.
+ * EnsureEnoughDiskSpaceForShardMove checks that there is enough space for
+ * shard moves of the given colocated shard list from source node to target node.
+ * It tries to clean up old shard placements to ensure there is enough space.
+ */
+static void
+EnsureEnoughDiskSpaceForShardMove(List *colocatedShardList,
+								  char *sourceNodeName, uint32 sourceNodePort,
+								  char *targetNodeName, uint32 targetNodePort)
+{
+	int64 colocationSizeInBytes = ColocationSizeInBytes(colocatedShardList,
+														sourceNodeName,
+														sourceNodePort);
+
+	uint32 connectionFlag = 0;
+	MultiConnection *connection = GetNodeConnection(connectionFlag, targetNodeName,
+													targetNodePort);
+	int checkSpaceTries = 0;
+	while (true)
+	{
+		bool droppedAllOldShards = false;
```

Comment:
What is the value that retrying to drop adds here? Can't we just try once and throw error if it fails?

That'd simplify the logic quite a bit and avoid 1 GUC: `WaitForDeferShardsMaxTries`?

Reply:
even 2 gucs I think: `WaitForDeferShardRetryTimeInSec`

Reply:
As we talked in private I think this doesn't add much value here. So simplifying makes sense, I am removing the 2 GUCs